dask_expr/io/parquet.py

```diff
@@ -111,26 +112,42 @@ def _maybe_adjust_cpu_count():
 
 class FragmentWrapper:
```

Comment:
This class is a bit wild but it's all well encapsulated. Once this stabilizes we can upstream certain things and the wrapper can fade out. API wise it's still a drop in replacement and this is merely there to facilitate serialization